socorro/lib/threaded_task_manager.py

```diff
@@ -8,68 +8,58 @@
-    )
-
     def __init__(
-        self, config, job_source_iterator=default_iterator, task_func=default_task_func
+        self,
+        idle_delay=7,
+        quit_on_empty_queue=False,
+        number_of_threads=4,
+        maximum_queue_size=8,
+        job_source_iterator=default_iterator,
+        task_func=default_task_func,
     ):
-        """the constructor accepts the function that will serve as the data
-        source iterator and the function that the threads will execute on
-        consuming the data.
-
-        parameters:
-            job_source_iterator - an iterator to serve as the source of data.
-                                  it can be of the form of a generator or
-                                  iterator; a function that returns an
-                                  iterator; a instance of an iterable object;
-                                  or a class that when instantiated with a
-                                  config object can be iterated.  The iterator
-                                  must yield a tuple consisting of a
-                                  function's tuple of args and, optionally, a
-                                  mapping of kwargs.
-                                  Ex:  (('a', 17), {'x': 23})
-            task_func - a function that will accept the args and kwargs yielded
-                        by the job_source_iterator"""
-        super().__init__(config, job_source_iterator, task_func)
+        """
+        :arg idle_delay: the delay in seconds if no job is found
+        :arg quit_on_empty_queue: stop if the queue is empty
+        :arg number_of_threads: number of worker threads to run
+        :arg maximum_queue_size: maximum size of the internal queue from which the
+            threads poll
+        :arg job_source_iterator: an iterator to serve as the source of data. it can
+            be of the form of a generator or iterator; a function that returns an
+            iterator; a instance of an iterable object; or a class that when
+            instantiated with a config object can be iterated. The iterator must
+            yield a tuple consisting of a function's tuple of args and, optionally,
+            a mapping of kwargs. Ex:  (('a', 17), {'x': 23})
+        :arg task_func: a function that will accept the args and kwargs yielded
+            by the job_source_iterator
+        """
+        super().__init__(
+            idle_delay=idle_delay,
+            quit_on_empty_queue=quit_on_empty_queue,
+            job_source_iterator=job_source_iterator,
+            task_func=task_func,
+        )
         self.thread_list = []  # the thread object storage
-        self.number_of_threads = config.number_of_threads
-        self.task_queue = queue.Queue(config.maximum_queue_size)
+        self.number_of_threads = number_of_threads
+        self.task_queue = queue.Queue(maximum_queue_size)
+
+        self.queueing_thread = None
```

Comment:
"queuing" turns out to be a valid spelling, but I hate it, so I changed it. Further, the `start` code creates this instance variable, but it should be initialized, so I added this line here.